parser/digester.go

```diff
@@ -193,6 +241,16 @@ func (d *sqlDigester) normalize(sql string, keepHint bool) {
 		}
 
 		d.reduceLit(&currTok)
+<<<<<<< HEAD:parser/digester.go
+=======
+		if forPlanReplayerReload {
+			// Apply for plan replayer to match specific rules, changing IN (...) to IN (?). This can avoid plan replayer load failures caused by parse errors.
+			d.replaceSingleLiteralWithInList(&currTok)
+		} else if forBinding {
+			// Apply binding matching specific rules, IN (?) => IN ( ... ) #44298
+			d.reduceInListWithSingleLiteral(&currTok)
```

Comment:
it seems change ??? to (...) is not existed in v7.1, maybe we can close this cherry-pick